thrust/detail/type_traits/has_trivial_assign.h

```diff
@@ -25,25 +25,19 @@
 #include <thrust/detail/config.h>
 #include <thrust/detail/type_traits.h>
 
+#include <cuda/std/type_traits>
+
 THRUST_NAMESPACE_BEGIN
 
 namespace detail
 {
 
-template<typename T> struct has_trivial_assign
+template<typename T> 
+struct has_trivial_assign
   : public integral_constant<
-      bool,
-      (is_pod<T>::value && !is_const<T>::value)
-#if THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_MSVC
-      || __has_trivial_assign(T)
-#elif THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_GCC
-// only use the intrinsic for >= 4.3
-#if (__GNUC__ >= 4) && (__GNUC_MINOR__ >= 3)
-      || __has_trivial_assign(T)
-#endif // GCC VERSION
-#elif THRUST_HOST_COMPILER == THRUST_HOST_COMPILER_CLANG
-      || __has_trivial_assign(T)
-#endif // THRUST_HOST_COMPILER
+      bool, 
```

Comment:
Ditto, can we just go with `is_trivially_copy_assignable`

Reply:
It's using different type, not `thrust::false_type` etc, so it would require replacing quite a few other places.